### PR TITLE
Create binary_search.cpp

### PR DESCRIPTION
**Pull request by Mayankjha997**
This is in c++ language.

### CODE REVIEW

**Comment by Mayankjha997** [comment]
can you help me in solving this issue

File: algorithms/c++/binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#include<stdlib.h>
+
+#define MAX_SIZE 5
+
+using namespace std;
+
+int main()
+{
+    int arr_search[MAX_SIZE], i, element;
+    int f = 0, r = MAX_SIZE, mid;
```
Comment (RavinduSachintha) [inline algorithms/c++/binary_search.cpp:12]: Codacy found an issue: [The scope of the variable 'mid' can be reduced.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6276107)

**Comment by RavinduSachintha** [comment]
> can you help me in solving this issue

sure :slightly_smiling_face: 
you are not allowed to create new folders. but you can create new files. We used **cpp** folders for c++ language. just make sure to put your file in that folder with a different name if there is already a file.

File: algorithms/c++/binary_search.cpp (hunk 1 above)
Comment (RavinduSachintha) [inline algorithms/c++/binary_search.cpp:12]: Codacy found an issue: [The scope of the variable 'mid' can be reduced.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6276107)

**Comment by RavinduSachintha** [comment]
@Mayankjha997 you have to delete the first changes (c++ folder) and make sure to use for the filename only "simple letters" & "underscore". (don't use numbers)

**Comment by Mayankjha997** [comment]
I can't find the way to delete plzz help me

**Comment by Mayankjha997** [comment]
Please check

**Comment by RavinduSachintha** [comment]
c++ folder changes have been deleted. now you just need to make sure the filename is right way. just follow the previously mentioned advice.

**Comment by Mayankjha997** [comment]
please review once more sir

**Comment by RavinduSachintha** [comment]
> please review once more sir
Don't create multiple files. delete the unnecessary one. Just use simple letters and underscore for file naming.

**Comment by Mayankjha997** [comment]
sir done

**Review by Danushka96 — DISMISSED** [review]

**Comment by Danushka96** [comment]
@Mayankjha997 Hi
What are the files you changed? You have removed all of your changes,

**Comment by Mayankjha997** [comment]
i have created B_inary_Search_exe.cpp in algorithm section plzz check

**Comment by Danushka96** [comment]
Hi @Mayankjha997 
make sure to use for the filename only "**simple letters**"

**Comment by Mayankjha997** [comment]
please suggest me which to used. i have used all

**Comment by Danushka96** [comment]
@Mayankjha997 try this `binary_search_exe.cpp`

**Comment by Mayankjha997** [comment]
sir This is the best which I can name . I have made my file in Algorithm/bsrh/Binary_search_exe.cpp .Please check

**Comment by Danushka96** [comment]
Hi @Mayankjha997 

FYI
Capital letters: "A, B, C, .."
Simple Letters: "a, b,c, .."

Filenames with capital letters are not allowed here
Binary_search_exe.cpp is wrong because B is capital as I mentioned above,
binary_search_exe.cpp is correct 

Hope you get it. Thanks

**Comment by Mayankjha997** [comment]
Done perfectly this time

**Comment by Mayankjha997** [comment]
@Danushka96 please check!!

**Comment by Mayankjha997** [comment]
is it done??

**Comment by RavinduSachintha** [comment]
> is it done??

sorry @Mayankjha997. Still we are not be able to merge your PR because it has failed on the compiling test. You have used some DOS based function "getch" from the `conio.h` header which is not a part of standard C++ libraries. Change it to use only standard C++ libraries (ex:- iostream ). Then your PR will be able to pass the test. Good luck !

**Comment by Mayankjha997** [comment]
sir @RavinduSachintha I think soo now its done!

**Comment by Mayankjha997** [comment]
@Danushka96 @RavinduSachintha I think it has passed all the test . soo you might now able to merge my PR

**Comment by RavinduSachintha** [comment]
> @Danushka96 @RavinduSachintha I think it has passed all the test . soo you might now able to merge my PR

Can you add your contribution to the main README file. you just need to copy and paste the "correct sign" phrase to the right place at the chart. Then we will be able to add your contribution for documentation too. No need to make another PR. just change and push it to this.

**Comment by Mayankjha997** [comment]
Sir i have added please review

File: README.md [hunk 2]
```diff
@@ -158,6 +158,7 @@ Only project maintainers should merge a PR. Other members can add their reviews
   <tr>
     <td align="center"><a href="https://github.com/Srinjoycode"><img src="https://avatars1.githubusercontent.com/u/55613633?v=4" width="100px;" alt=""/><br /><sub><b>Srinjoy Bhuiya</b></sub></a><br /><a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=Srinjoycode" title="Code">💻</a> <a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=Srinjoycode" title="Documentation">📖</a></td>
     <td align="center"><a href="https://github.com/ashera96"><img src="https://avatars3.githubusercontent.com/u/30475839?v=4" width="100px;" alt=""/><br /><sub><b>Ashera Silva</b></sub></a><br /><a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=ashera96" title="Code">💻</a> <a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=ashera96" title="Documentation">📖</a></td>
+    <td align="center"><a href="https://github.com/Mayankjha997"><img src="https://avatars2.githubusercontent.com/u/50633678?v=4" width="100px;" alt=""/><br /><sub><b>Mayank997</b></sub></a><br /><a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=Mayankjha997" title="Code">💻</a></td>
```
Comment (RavinduSachintha) [inline README.md:161]: not this section. you need to update the algorithm chart data. please make sure to revert this change

**Comment by Mayankjha997** [comment]
sir can you please specify. i can't find where

**Comment by RavinduSachintha** [comment]
@Mayankjha997 if you feel okay to merge now, I can do it now. I will add the documentation. Is it okay?

**Comment by Mayankjha997** [comment]
yes @RavinduSachintha  sir sure

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@all-contributors please add @Mayankjha997 for code
Thank you @Mayankjha997  for your contribution to this repository. :+1: 
Make sure to star the repository. :wink:
